PR: https://github.com/odoo/odoo/pull/107504

From: 999043b36c123972cb10dd1c55ce71a3cc981f85
From: niyasraphy
Date: 2022-12-09 15:40:58

Structural Changes: 6
Total Changes: 12

[IMP] *: change leave string's to time off

update various string/labels contains "Leave" in hr_holidays, hr_work_entry_contract, resource, test_resource and project_timesheet_holidays modules to "Time Off"

closes odoo/odoo#107504

Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_holidays/models/hr_leave.py
+++ b/addons/hr_holidays/models/hr_leave.py
@@ -35,7 +35,7 @@ def get_employee_from_context(values, context, user_employee_id):
     return employee_id_value or context.get('default_employee_id', context.get('employee_id', user_employee_id))
 
 class HolidaysRequest(models.Model):
-    """ Leave Requests Access specifications
+    """ Time Off Requests Access specifications
 
      - a regular employee / user
       - can see all leaves;
@@ -1454,7 +1454,7 @@ class HolidaysRequest(models.Model):
                         raise UserError(_('You must either be a Time off Officer or Time off Manager to approve this leave'))
 
     ###################################################
-    # Leave modification methods
+    # Time Off modification methods
     ###################################################
 
     def _split_leave_on_gto(self, gto): #gto = global time off

--- a/addons/hr_holidays/models/hr_leave_allocation.py
+++ b/addons/hr_holidays/models/hr_leave_allocation.py
@@ -134,7 +134,7 @@ class HolidaysAllocation(models.Model):
     is_officer = fields.Boolean(compute='_compute_is_officer')
     accrual_plan_id = fields.Many2one('hr.leave.accrual.plan', compute="_compute_from_holiday_status_id", store=True, readonly=False, domain="['|', ('time_off_type_id', '=', False), ('time_off_type_id', '=', holiday_status_id)]", tracking=True)
     max_leaves = fields.Float(compute='_compute_leaves')
-    leaves_taken = fields.Float(compute='_compute_leaves')
+    leaves_taken = fields.Float(compute='_compute_leaves', string='Time off Taken')
     taken_leave_ids = fields.One2many('hr.leave', 'holiday_allocation_id', domain="[('state', 'in', ['confirm', 'validate1', 'validate'])]")
 
     _sql_constraints = [

--- a/addons/hr_holidays/models/hr_leave_type.py
+++ b/addons/hr_holidays/models/hr_leave_type.py
@@ -68,7 +68,7 @@ class HolidaysType(models.Model):
         ('no_validation', 'No Validation'),
         ('hr', 'By Time Off Officer'),
         ('manager', "By Employee's Approver"),
-        ('both', "By Employee's Approver and Time Off Officer")], default='hr', string='Leave Validation')
+        ('both', "By Employee's Approver and Time Off Officer")], default='hr', string='Time Off Validation')
     requires_allocation = fields.Selection([
         ('yes', 'Yes'),
         ('no', 'No Limit')], default="yes", required=True, string='Requires allocation',

--- a/addons/hr_holidays/models/resource.py
+++ b/addons/hr_holidays/models/resource.py
@@ -8,7 +8,7 @@ from odoo.osv import expression
 class CalendarLeaves(models.Model):
     _inherit = "resource.calendar.leaves"
 
-    holiday_id = fields.Many2one("hr.leave", string='Leave Request')
+    holiday_id = fields.Many2one("hr.leave", string='Time Off Request')
 
     @api.constrains('date_from', 'date_to', 'calendar_id')
     def _check_compare_dates(self):
@@ -105,7 +105,7 @@ class CalendarLeaves(models.Model):
 class ResourceCalendar(models.Model):
     _inherit = "resource.calendar"
 
-    associated_leaves_count = fields.Integer("Leave Count", compute='_compute_associated_leaves_count')
+    associated_leaves_count = fields.Integer("Time Off Count", compute='_compute_associated_leaves_count')
 
     def _compute_associated_leaves_count(self):
         leaves_read_group = self.env['resource.calendar.leaves'].read_group(
